### PR TITLE
Fix DeepSpeed lr scheduler logic

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Fixes https://github.com/PyTorchLightning/pytorch-lightning/issues/8520

### Does your PR introduce any breaking changes ? If yes, please list them.

None

## Before submitting
- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [x] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you **update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)**? (not for typos, docs, test updates, or internal minor changes/refactorings)
- [x] Did you list all the **breaking changes** introduced by this pull request?

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 1]
```diff
@@ -387,45 +389,51 @@ def init_deepspeed(self):
...
+
         self._set_deepspeed_activation_checkpointing()
 
-        # set optimizer for save/load, but deepspeed manages the specific optimizer logic
-        self.lightning_module.trainer.optimizers = [optimizer]
-        self.lightning_module.trainer.schedulers = [lr_scheduler]
+        # although we set these here, deepspeed manages the specific optimizer logic
+        self.lightning_module.trainer.optimizers = [deepspeed_optimizer]
+        lr_scheduler["scheduler"] = deepspeed_scheduler
+        self.lightning_module.trainer.lr_schedulers = [lr_scheduler]
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/deepspeed.py:436]: This is the main fix of the PR

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 2]
```diff
@@ -459,13 +467,14 @@ def _set_deepspeed_activation_checkpointing(self):
...
     def _initialize_deepspeed_inference(self, model):
         # todo: Currently DeepSpeed requires optimizers at inference to partition weights correctly
-        optimizer, lightning_scheduler, optimizer_frequencies = None, None, None
+        optimizer, scheduler = None, None
         if "optimizer" not in self.config:
             rank_zero_info(
                 "You have not specified an optimizer or scheduler within the DeepSpeed config."
                 "Using `configure_optimizers` to define optimizer and scheduler."
             )
-            optimizer, lightning_scheduler, optimizer_frequencies = self._init_scheduler_optimizer()
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/deepspeed.py:468]: @SeanNaren there is no use for `optimizer_frequency`, correct?

File: pytorch_lightning/tuner/batch_size_scaling.py [hunk 3]
```diff
@@ -121,7 +121,7 @@ def __scale_batch_reset_params(trainer: 'pl.Trainer', model: 'pl.LightningModule
     trainer.logger = DummyLogger()
     trainer.callbacks = []  # not needed before full run
     trainer.limit_train_batches = 1.0
-    trainer.optimizers, trainer.schedulers = [], []  # required for saving
+    trainer.optimizers, trainer.lr_schedulers = [], []  # required for saving
```
Comment (carmocca) [inline pytorch_lightning/tuner/batch_size_scaling.py:124]: my grep powers found that these are also wrong - but unrelated to the DeepSpeed fixes in this PR

File: pytorch_lightning/plugins/training_type/deepspeed.py [hunk 4]
```diff
@@ -459,13 +467,14 @@ def _set_deepspeed_activation_checkpointing(self):
 
     def _initialize_deepspeed_inference(self, model):
```
Comment (carmocca) [inline pytorch_lightning/plugins/training_type/deepspeed.py:468]: @SeanNaren does this method not need to set the trainer `optimizer`/`lr_schedulers` attributes at the end? just like we do for `_initialize_deepspeed_train`

**Review by awaelchli — APPROVED** [review]

**Review by Borda — APPROVED** [review]

**Review by kaushikb11 — APPROVED** [review]

**Review by SeanNaren — APPROVED** [review]

**Comment by SeanNaren** [comment]
Thanks @carmocca!
